src/Entity/IdeasWorkshop/Idea.php

```diff
@@ -0,0 +1,234 @@
+        UuidInterface $uuid,
+        string $name,
+        Adherent $adherent,
+        Category $category,
+        Theme $theme,
+        ?Committee $committee,
+        ?\DateTime $publishedAt = null,
+        string $status = IdeaStatusEnum::IN_PROGRESS
+    ) {
+        $this->uuid = $uuid;
+        $this->setName($name);
+        $this->adherent = $adherent;
+        $this->category = $category;
+        $this->theme = $theme;
+        $this->committee = $committee;
+        $this->publishedAt = $publishedAt;
+        $this->status = $status;
+        $this->needs = new ArrayCollection();
+        $this->answers = new ArrayCollection();
+    }
+
+    public static function createUuid(string $name): UuidInterface
+    {
+        return Uuid::uuid5(Uuid::NAMESPACE_OID, $name);
+    }
+
+    public function getTheme(): Theme
+    {
+        return $this->theme;
+    }
+
+    public function setTheme(Theme $theme): void
+    {
+        $this->theme = $theme;
+    }
+
+    public function getCategory(): Category
+    {
+        return $this->category;
+    }
+
+    public function setCategory(Category $category): void
+    {
+        $this->category = $category;
+    }
+
+    public function getNeeds(): ArrayCollection
+    {
+        return $this->needs;
+    }
+
+    public function addNeed(Need $need): void
+    {
+        if (!$this->needs->contains($need)) {
+            $this->needs->add($need);
+        }
+    }
+
+    public function removeNeed(Need $need): void
+    {
+        $this->needs->removeElement($need);
+    }
+
+    public function getAdherent(): Adherent
+    {
+        return $this->adherent;
+    }
+
+    public function setAdherent(Adherent $adherent): void
+    {
+        $this->adherent = $adherent;
+    }
+
+    public function getPublishedAt(): \DateTime
+    {
+        return $this->publishedAt;
+    }
+
+    public function setPublishedAt(\DateTime $publishedAt): void
+    {
+        $this->publishedAt = $publishedAt;
+    }
+
+    public function getCommittee(): Committee
+    {
+        return $this->committee;
+    }
+
+    public function setCommittee(Committee $committee): void
+    {
+        $this->committee = $committee;
+    }
+
+    public function getStatus(): string
+    {
+        return $this->status;
+    }
+
+    public function setStatus(string $status): void
+    {
+        $this->status = $status;
+    }
+
+    public function addAnswer(Answer $answer): void
+    {
+        if (!$this->answers->contains($answer)) {
+            $this->answers->add($answer);
+            $answer->setIdea($this);
+        }
+    }
+
+    public function removeAnswer(Answer $answer): void
+    {
+        $this->answers->removeElement($answer);
+    }
+
+    public function getAnswers(): ArrayCollection
+    {
+        return $this->answers;
+    }
+
+    public function getDaysBeforeDeadline(): int
+    {
+        $deadline = $this->createdAt->add(new \DateInterval(self::PUBLISHED_INTERVAL));
```

Comment:
you need to initialize `$this->createdAt` in `__construct` method. Actually it's always null.

Reply:
createdAt is a property from the `EntityTimestampableTrait` and is set with the `@Gedmo\Timestampable(on="create")` why should I init it in the construct method ?